frame/evm/precompile/contracts/encoder/src/lib.rs

```diff
@@ -47,29 +57,93 @@ where
 		_target_gas: Option<u64>,
 		_context: &Context,
 	) -> core::result::Result<PrecompileOutput, ExitError> {
-		if input.len() < SELECTOR {
+		if input.len() < ACTION_LEN {
 			return Err(ExitError::Other("input length less than 4 bytes".into()));
 		}
-		let selector = &input[0..SELECTOR];
-		let call: T::Call = match selector {
-			_ if selector == <from_substrate_issuing::Pallet<T>>::digest() => {
-				from_substrate_issuing::Call::<T>::asset_burn_event_handle(input.to_vec()).into()
+		let pallet_digest = &input[0..PALLET_DIG_LEN];
+		let method_digest = &input[PALLET_DIG_LEN..ACTION_LEN];
+		let output = match pallet_digest {
+			_ if pallet_digest == <from_substrate_issuing::Pallet<T>>::digest() => {
+				match method_digest {
+					_ if method_digest
+						== &sha3::Keccak256::digest(BURN_AND_REMOTE_UNLOCK_METHOD)
+							[..METHOD_DIG_LEN] =>
```

Comment:
Can we do like the following:

```rs
let digest = |method| { sha3::Keccak256::digest(method) }[..METHOD_DIG_LEN];

_ if method_digest == digest(BURN_AND_REMOTE_UNLOCK_METHOD) => {...}
```

Reply:
I have tried before, but it reports an unknown size error. Any idea?

Reply:
I don't know the context.

But basically, there are two ways.

1. reference the return type
2. specify the return type, IMO the `fn digest` could return a generic

Reply:
I have tried this way in my local, it works. FYI

```rs
let digest = |param_method, expected_method| -> bool {
	&sha3::Keccak256::digest(param_method)[..METHOD_DIG_LEN] == expected_method
};

_ if digest(method_digest, BURN_AND_REMOTE_UNLOCK_METHOD) => {..}
```

Reply:
If that then rename to `check_digest`

Reply:
Right.

Reply:
This way looks good.